src/frontend/mosh-server.cc

```diff
@@ -554,7 +554,24 @@ static int run_server( const char *desired_ip, const char *desired_port,
     chdir_homedir();
 
     if ( with_motd && (!motd_hushed()) ) {
-      print_motd();
+      // For Ubuntu, try and print {,/var}/run/motd.dynamic.
+      // If they're the same, only print once.
+      // This file is only updated when pam_motd is run, but when
+      // mosh-server is run in the usual way with ssh via the script,
+      // this always happens.
+      // XXX Hackish knowledge of Ubuntu PAM configuration.
+      // But this seems less awful than build-time detection with autoconf.
+      struct stat first;
+      struct stat second;
+      if (!print_motd("/run/motd.dynamic") ||
+	  (stat("/run/motd.dynamic", &first) < 0 &&
+	   stat("/var/run/motd.dynamic", &second) < 0 &&
```

Comment:
I think you meant for these `<`s to be `==`s, or something. Though is this extra complexity needed? PAM isn’t reasonably going to be updating both `/run/motd.dynamic` and `/var/run/motd.dynamic` if they’re different files.